packages/components/PersonaCoin/src/PersonaCoin.tokens.ring.ts

```diff
@@ -0,0 +1,64 @@
+import { styleFunction } from '@uifabricshared/foundation-tokens';
+import { ViewProps } from 'react-native';
+import { IPersonaCoinTokens } from './PersonaCoin.types';
+import { ITheme } from '@uifabricshared/theming-ramp';
+import { getRingThickness, calculateEffectiveSizes } from './PersonaCoin.helpers';
+
+const _ringProps: (keyof IPersonaCoinTokens)[] = ['coinSize', 'size', 'ring'];
+
+function _buildRingStyles(tokenProps: IPersonaCoinTokens, theme: ITheme): ViewProps {
+  const { physicalSize } = calculateEffectiveSizes(tokenProps);
+  const { ring } = tokenProps;
+
+  if (!ring) return {};
+
+  const innerGap = ring.innerGap != undefined ? getRingThickness(ring.innerGap) : getRingThickness(ring.ringThickness || 'xxlarge');
+
+  const effectiveRingThickness = 2 * getRingThickness(ring.ringThickness || 'xxlarge') + innerGap;
+  const effectiveSize = physicalSize + 2 * effectiveRingThickness;
+  const ringColor = ring.transparent ? 'transparent' : ring.ringBackgroundColor || theme.colors.personaring;
+
+  return {
+    style: {
+      borderStyle: 'solid',
+      borderColor: ringColor,
+      borderWidth: effectiveRingThickness,
+      borderRadius: effectiveSize / 2,
+      width: effectiveSize,
+      height: effectiveSize,
+      position: 'absolute',
+      top: -effectiveRingThickness,
+      left: -effectiveRingThickness,
+    },
+  };
+}
+
+function _buildGlowStyles(tokenProps: IPersonaCoinTokens, theme: ITheme): ViewProps {
```

Comment:
why the leading underscore?

Reply:
this was something I saw in other tokens.xx.ts files. so maintained uniformity here too !

Reply:
looks like it's a convention for functions that get transformed by the StyleFunction at the end. Interesting...